### ISSUE TITLE
macos menu cannot expand, keep flickers 

### ISSUE THREAD

**Issue by jizhi0v0** [zed-industries/zed#20710]
### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

1.click the zed's menu

### Environment

zed preview 0.162.1 version
Apple M3 Max 
36 GB
15.1 (24B83)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```
[Zed.log](https://github.com/user-attachments/files/17770215/Zed.log)

```
</details>

**Comment by jizhi0v0** [comment]
Performance video
https://github.com/user-attachments/assets/982266ca-4ad6-4ae1-bb1f-e56addc14ec1

**Comment by mrnugget** [comment]
I'm on macOS 15.0.1 and not 15.1, cannot reproduce. Do you have any hunch as to whether this tarted to show up with the update to 15.1?

**Comment by jizhi0v0** [comment]
> I'm on macOS 15.0.1 and not 15.1, cannot reproduce. Do you have any hunch as to whether this tarted to show up with the update to 15.1?

I'm not sure. I don't have another one

**Comment by opsnull** [comment]
same problem here.

Zed: v0.163.0 (Zed Dev 332b33716aa9595edcb6e0c394b4b7db137ef496)
OS: macOS 15.1.0
Memory: 18 GiB
Architecture: aarch64

**Comment by SomeoneToIgnore** [comment]
Has to be something else than macOS 15.1 as I do not see this:

https://github.com/user-attachments/assets/77a2bac6-2c20-41bd-8ccb-1927c14e0c4e

**Comment by jizhi0v0** [comment]
> Has to be something else than macOS 15.1 as I do not see this:
> 
>  no_flicker.mov

I have tested 0.162.1 and 0.162.0 version.They all have this issue.

0.161.1 version can be used normally.

**Comment by jizhi0v0** [comment]
still in 0.162.2

**Comment by mrnugget** [comment]
Since I cannot reproduce this: is somehow who can reproduce it willing to run a `git bisect` between these two tags (https://github.com/zed-industries/zed/compare/v0.161.1...v0.162.0-pre) and see which commit introduced the regression? I bet it can even be narrowed down to only changes in the `gpui` crate.

**Comment by mindon** [comment]
same issue with zed 0.162.3, under macos sequoia 15.1

**Comment by mindon** [comment]
> > same issue with zed 0.162.3, under macos sequoia 15.1
> 
> Have you used https://github.com/Kyle-Ye/XcodeLLMEligible project?
> 
> Wait...Are all the problems with the China SKU?

have no idea, it works right now, without anything updated -- duration these 30 minutes, a few projects opened and closed

**Comment by jizhi0v0** [comment]
> Since I cannot reproduce this: is somehow who can reproduce it willing to run a `git bisect` between these two tags ([v0.161.1...v0.162.0-pre](https://github.com/zed-industries/zed/compare/v0.161.1...v0.162.0-pre)) and see which commit introduced the regression? I bet it can even be narrowed down to only changes in the `gpui` crate.

I had do it complet.Hope this helps you

ff4f67993b5da8d2444c88c851e35614fe7682ba is the first bad commit
commit ff4f67993b5da8d2444c88c851e35614fe7682ba
Author: Conrad Irwin <conrad.irwin@gmail.com>
Date:   Fri Nov 8 13:05:10 2024 -0700

    macOS: Add key equivalents for non-Latin layouts (#20401)
    
    Closes  #16343
    Closes #10972
    
    Release Notes:
    
    - (breaking change) On macOS when using a keyboard that supports an
    extended Latin character set (e.g. French, German, ...) keyboard
    shortcuts are automatically updated so that they can be typed without
    `option`. This fixes several long-standing problems where some keyboards
    could not type some shortcuts.
    - This mapping works the same way as
    [macOS](https://developer.apple.com/documentation/swiftui/view/keyboardshortcut(_:modifiers:localization:)).
    For example on a German keyboard shortcuts like `cmd->` become `cmd-:`,
    `cmd-[` and `cmd-]` become `cmd-ö` and `cmd-ä`. This mapping happens at
    the time keyboard layout files are read so the keybindings are visible
    in the command palette. To opt out of this behavior for your custom
    keyboard shortcuts, set `"use_layout_keys": true` in your binding
    section. For the mappings used for each layout [see
    here](https://github.com/zed-industries/zed/blob/a890df1863ca939ee7f0ada6e629b6f83eb18bc5/crates/settings/src/key_equivalents.rs#L7).
    
    ---------
    
    Co-authored-by: Will <will@zed.dev>

 Cargo.toml                                   |   1 +
 assets/keymaps/vim.json                      |  27 ++++
 crates/gpui/examples/input.rs                |  11 +-
 crates/gpui/src/app.rs                       |  39 ++++++
 crates/gpui/src/keymap/binding.rs            |  23 +++-
 crates/gpui/src/platform.rs                  |   2 +
 crates/gpui/src/platform/linux/platform.rs   |   8 ++
 crates/gpui/src/platform/mac/platform.rs     |  62 +++++++++
 crates/gpui/src/platform/test/platform.rs    |   6 +
 crates/gpui/src/platform/windows/platform.rs |   8 ++
 crates/settings/src/key_equivalents.rs       | 185 +++++++++++++++++++++++++++
 crates/settings/src/keymap_file.rs           |  24 +++-
 crates/settings/src/settings.rs              |   1 +
 crates/workspace/src/workspace.rs            |   1 +
 crates/zed/src/zed.rs                        |   9 +-
 docs/src/key-bindings.md                     |  38 +++++-
 16 files changed, 435 insertions(+), 10 deletions(-)
 create mode 100644 crates/settings/src/key_equivalents.rs

**Comment by jizhi0v0** [comment]
> > Since I cannot reproduce this: is somehow who can reproduce it willing to run a `git bisect` between these two tags ([v0.161.1...v0.162.0-pre](https://github.com/zed-industries/zed/compare/v0.161.1...v0.162.0-pre)) and see which commit introduced the regression? I bet it can even be narrowed down to only changes in the `gpui` crate.
> 
> I saw this submission and I know why, because my Chinese input method, if I switch to English input method, there will be no such problem

If you need to reproduce this problem, you can install https://shurufa.sogou.com/ 

> 
> [ff4f679](https://github.com/zed-industries/zed/commit/ff4f67993b5da8d2444c88c851e35614fe7682ba) is the first bad commit commit [ff4f679](https://github.com/zed-industries/zed/commit/ff4f67993b5da8d2444c88c851e35614fe7682ba) Author: Conrad Irwin [conrad.irwin@gmail.com](mailto:conrad.irwin@gmail.com) Date: Fri Nov 8 13:05:10 2024 -0700
> 
> ```
> macOS: Add key equivalents for non-Latin layouts (#20401)
> 
> Closes  #16343
> Closes #10972
> 
> Release Notes:
> 
> - (breaking change) On macOS when using a keyboard that supports an
> extended Latin character set (e.g. French, German, ...) keyboard
> shortcuts are automatically updated so that they can be typed without
> `option`. This fixes several long-standing problems where some keyboards
> could not type some shortcuts.
> - This mapping works the same way as
> [macOS](https://developer.apple.com/documentation/swiftui/view/keyboardshortcut(_:modifiers:localization:)).
> For example on a German keyboard shortcuts like `cmd->` become `cmd-:`,
> `cmd-[` and `cmd-]` become `cmd-ö` and `cmd-ä`. This mapping happens at
> the time keyboard layout files are read so the keybindings are visible
> in the command palette. To opt out of this behavior for your custom
> keyboard shortcuts, set `"use_layout_keys": true` in your binding
> section. For the mappings used for each layout [see
> here](https://github.com/zed-industries/zed/blob/a890df1863ca939ee7f0ada6e629b6f83eb18bc5/crates/settings/src/key_equivalents.rs#L7).
> 
> ---------
> 
> Co-authored-by: Will <will@zed.dev>
> ```
> 
> Cargo.toml | 1 + assets/keymaps/vim.json | 27 ++++ crates/gpui/examples/input.rs | 11 +- crates/gpui/src/app.rs | 39 ++++++ crates/gpui/src/keymap/binding.rs | 23 +++- crates/gpui/src/platform.rs | 2 + crates/gpui/src/platform/linux/platform.rs | 8 ++ crates/gpui/src/platform/mac/platform.rs | 62 +++++++++ crates/gpui/src/platform/test/platform.rs | 6 + crates/gpui/src/platform/windows/platform.rs | 8 ++ crates/settings/src/key_equivalents.rs | 185 +++++++++++++++++++++++++++ crates/settings/src/keymap_file.rs | 24 +++- crates/settings/src/settings.rs | 1 + crates/workspace/src/workspace.rs | 1 + crates/zed/src/zed.rs | 9 +- docs/src/key-bindings.md | 38 +++++- 16 files changed, 435 insertions(+), 10 deletions(-) create mode 100644 crates/settings/src/key_equivalents.rs

**Comment by mindon** [comment]
@jizhi0v0 yes! this happens under macos 15.1 when input source switching to **Pinyin-Simplified**

**Comment by mrnugget** [comment]
Thank you @jizhi0v0! That's very helpful.

@ConradIrwin I'm out of my depth here — any hunches?

**Comment by ConradIrwin** [comment]
Looks like the Pinyin keyboard switches to US when you open the menu, which causes us to re-load the keymap, which causes us to re-render the menu, which causes the menu to close.

As the Pinyin keyboard does not require any key equivalents, we can skip this whole step.

Fix is #20914
